src/OnchainKitProvider.tsx

```diff
@@ -10,16 +10,24 @@ export const OnchainKitContext = createContext<OnchainKitContextType>(ONCHAIN_KI
  */
 export function OnchainKitProvider({
   address,
+  apiKey,
   chain,
   children,
   schemaId,
 }: OnchainKitProviderProps) {
   if (schemaId && !checkHashLength(schemaId, 64)) {
     throw Error('EAS schemaId must be 64 characters prefixed with "0x"');
   }
+  if (!apiKey) {
+    throw Error(
+      'API Key is required to use OnchainKit - get your API key from the Coinbase Developer Platform: https://portal.cdp.coinbase.com/products/templates',
+    );
+  }
   const value = useMemo(() => {
     return {
       address: address ?? null,
+      apiKey: apiKey,
+      rpcUrl: `https://api.developer.coinbase.com/rpc/v1/${ONCHAIN_KIT_CONFIG.chain.name.replace(' ', '-').toLowerCase()}/${ONCHAIN_KIT_CONFIG.apiKey}`,
```

Comment:
So this means the `rpcUrl` is always CDP related? 🤔 what if I want something custom, and I want to use OnchainKit with my personal RPC.

Also all like 30 should be `getRpcUrl`, this all interpolation and mixing is enough logic that should be is all function.

Reply:
added default value for `rpcUrl` (if not provided and if API key is set)